osu.Game/Rulesets/Objects/BarLineGenerator.cs

```diff
@@ -46,6 +47,16 @@ public BarLineGenerator(IBeatmap beatmap)
 
                 for (double t = currentTimingPoint.Time; Precision.DefinitelyBigger(endTime, t); t += barLength, currentBeat++)
                 {
+                    var roundedTime = Math.Round(t, MidpointRounding.AwayFromZero);
+
+                    // in the case of some bar lengths, rounding errors can cause t to be slightly less than
+                    // the expected whole number value due to floating point inaccuracies.
+                    // if this is the case, apply rounding.
+                    if (Precision.AlmostEquals(t, roundedTime))
```

Comment:
Not really sure about this. It's probably not going to come up as an issue but we're talking about less than 0.5ms difference here, right? Something like that isn't going to noticeable and I say this check shouldn't exist to prevent potential issues as a result of it existing.

Reply:
Well, at the current state, it'll only catch differences to the tune of 1e-6 ms or less, as per the default tolerance.

I'm not sure how this can be applied more centrally. I also think more aggressive rounding feels incorrect.

Reply:
Always rounding being the alternative? I swear I had a similar review comment on this PR but must have gotten misplaced.

Reply:
I suppose this is fine given that these are doubles, from a simple test you'd need ~1000000 bar lines to cause a failure here. Let's go with this for now.